core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCTemplate.scala

```diff
@@ -0,0 +1,296 @@
+package org.bitcoins.core.protocol.dlc
+
+import org.bitcoins.core.currency._
+import org.bitcoins.core.protocol.dlc.ContractOraclePair._
+import org.bitcoins.core.protocol.dlc.DLCTemplate.validateMatchingOracleDescriptors
+import org.bitcoins.core.protocol.tlv._
+
+sealed trait DLCTemplate {
+  def totalCollateral: CurrencyUnit
+
+  def oracles: Vector[OracleAnnouncementTLV]
+
+  def oracleThreshold: Int
+
+  def contractDescriptor: ContractDescriptor
+
+  def oracleInfo: OracleInfo
+
+  def toContractInfo: ContractInfo
+
+  def individualCollateral: CurrencyUnit =
+    totalCollateral.satoshis / Satoshis(2)
+}
+
+sealed trait SingleOracleDLCTemplate extends DLCTemplate {
+  def oracle: OracleAnnouncementTLV
+
+  override def oracleInfo: SingleOracleInfo
+
+  final override def oracleThreshold: Int = 1
+  final override def oracles: Vector[OracleAnnouncementTLV] = Vector(oracle)
+}
+
+sealed trait MultiOracleDLCTemplate extends DLCTemplate {
+  override def oracleInfo: MultiOracleInfo[SingleOracleInfo]
+}
+
+sealed trait CFDTemplate extends DLCTemplate {
+  override def oracleInfo: NumericOracleInfo
+
+  def currentPrice: Long
```

Comment:
for uniformity this might be better named the strike price ... I think? Right @Christewart ?